compiler/src/iree/compiler/Codegen/Common/OptimizeVectorTransferPass.cpp

```diff
@@ -122,7 +122,8 @@ struct OptimizeVectorTransferPass
     }
 
     // Second stage of patterns to flatten transfer ops.
-    {
+    auto variantOp = getExecutableVariantOp(funcOp);
+    if (!variantOp.hasValue() || !isSPIRV(variantOp.value())) {
```

Comment:
I think the same problem could happen for cuda or any other backend relying on 2d matmul kind of ops. I would rather either make it an explicit option of the pass or move it to a separate pass.

Reply:
Done.